app/validation.py

```diff
@@ -405,3 +405,11 @@ def admin_email_address_has_approved_domain(email_address):
     :return: boolean
     """
     return email_address.split('@')[-1] in current_app.config.get('DM_ALLOWED_ADMIN_DOMAINS', [])
+
+
+def is_valid_email_address(email_address: str) -> bool:
+    "Check the email address is valid"
+    # regex from Mozilla Developer Network
+    # https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#Validation
+    pattern = r"^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"  # noqa: E501
```

Comment:
What do we think is the risk of an existing email address in our DB (perhaps entered by mistake) not matching this regex? (I would say low, and the likelihood of the user retyping the mistake when e.g. resetting a pw probably lower...?)

Reply:
Presumably any that didn't match wouldn't be a valid email address and therefore would be an inactive account?

Reply:
Unless something in our email sending pipeline silently stripped nulls - I think it's worth checking before rollout...

Reply:
It should be easy to check using `api-clients-shell.py` on a connection to production...

Reply:
As long as this is more strict than:
https://trello.com/c/3plghQzC/427-our-wtforms-email-validation-is-less-strict-than-notifys
https://github.com/alphagov/notifications-utils/blob/67889886ec1476136d12e7f32787a7dbd0574cc2/notifications_utils/recipients.py#L439

All good 👍

Reply:
Currently running a test against production

```
import re
email_address_pattern = ...
invalid_email_addresses = [print(user["id"]) or user for user in data.find_users_iter() if not user["personalDataRemoved"] and not re.fullmatch(email_address_pattern, user["emailAddress"])]
```

Reply:
Looks like there aren't any active users whose email address don't fit the pattern

Reply:
:100:

Reply:
Nice one 👍